Sources/ComposableArchitecture/Store.swift

```diff
@@ -348,21 +348,25 @@ public final class Store<State, Action> {
         guard !isSending else { return }
 
         let callbackInfo = Instrumentation.CallbackInfo<Self.Type, Any>(storeKind: Self.self, action: nil).eraseToAny()
-        instrumentation.callback?(callbackInfo, .pre, .storeToLocal)
+        instrumentation.callback?(callbackInfo, .pre, .scopedStoreToLocal)
         let newLocalState = toLocalState(newValue)
-        instrumentation.callback?(callbackInfo, .post, .storeToLocal)
+        instrumentation.callback?(callbackInfo, .post, .scopedStoreToLocal)
 
         guard let previousState = localStore?.state.value, let isDuplicate = isDuplicate else {
+            instrumentation.callback?(callbackInfo, .pre, .scopedStoreChangeState)
```

Comment:
At this point this is getting pretty silly. I wish that Swift had as clean a scoping mechanism as C++ and we could use RAII with a return type from the callback that on destruction out of scope called the `.post` instance.

Reply:
Yeah we have that too. And in this guard it would work. But there are a few others where to add the scope increases total line count and indentation. Which is sad.